src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -366,9 +411,33 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                     scene.pointerY = this._screenCoordinatesRef.y;
                 }
             }
-            controllerData.pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
 
-            const pick = controllerData.pick;
+            let hover = this._pickWithMesh(controllerData.hoverMesh, false);
+            let nearHover = hover && hover.pickedPoint && hover.hit;
+            if(nearHover)
+            {
+                console.log("HOVER SUCCESS");
+                controllerData.hover = hover;
+                controllerData.nearHover = true;
+            }
+
+            let pick = this._pickWithMesh(controllerData.pickMesh, true);
+            let nearPick = pick && pick.pickedPoint && pick.hit;
+ 
+            if (nearPick) {
+                console.log("PICK SUCCESS");
+                controllerData.laserPointer.isVisible = false;
+                controllerData.nearPick = true;
+            }
+            else {
+                pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
+                if (pick && pick.pickedPoint && pick.hit) {
+                    // Update laser state
+                    this._updatePointerDistance(controllerData.laserPointer, pick.distance);
+                }
+            }
+     
+            controllerData.pick = pick;
 
             if (pick && pick.pickedPoint && pick.hit) {

```

Comment:
There's some logic in here specific to the laser that shows up for far interaction that we shouldn't be doing if we have a near interaction.

Reply:
so, it should be added?

Reply:
moved it to only apply for far interaction

Reply:
There's still ray-specific logic being called no matter what below

Reply:
I see, i did not know selection Mesh is only supposed to be populated for ray. Do you think none of this applies to near interaction at all?